File: android/app/src/main/java/jaclib/peer/PeerReference.java
package jaclib.peer;


import java.lang.ref.WeakReference;

public abstract class PeerReference extends WeakReference {

    public PeerReference b;

    public PeerReference a;

    public long peer;

    protected PeerReference(Peer arg0, ti arg1) {
        super(arg0, arg1.b);
        arg1.a(this, (byte) 92);
    }

    public void setPeer(long arg0) {
        this.b(0);
        this.peer = arg0;
    }

    public final boolean a(int arg0) {
        return this.peer != 0L;
    }

    public final long b(int arg0) {
        long local12;
        if (this.peer == 0L) {
            local12 = 0L;
        } else {
            local12 = this.releasePeer(this.peer);
            this.peer = 0L;
        }
        return local12;
    }

    protected abstract long releasePeer(long arg0);
}

File: src/jvmMain/kotlin/jaclib/peer/PeerReference.kt
package jaclib.peer

import java.lang.ref.WeakReference


abstract class PeerReference protected constructor(arg0: Peer?, arg1: hb?) : WeakReference<Any?>(arg0, arg1?.a) {
    var a: PeerReference? = null

    private var peer: Long = 0

    var b: PeerReference? = null

    init {
        arg1?.a(this, -22143)
    }

    protected abstract fun releasePeer(arg0: Long): Long

    fun a(arg0: Byte): Boolean {
        return this.peer != 0L
    }

    private fun setPeer(arg0: Long) {
        this.a(false)
        this.peer = arg0
    }

    fun a(arg0: Boolean): Long {
        val local11: Long
        if (this.peer == 0L) {
            local11 = 0L
        } else {
            local11 = this.releasePeer(this.peer)
            this.peer = 0L
        }
        return local11
    }
}
